### PR TITLE
ON-43977 # Added comprehensive validation errors toast

### PR DESCRIPTION
**Pull request by Zaxist**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [x] Have you tested your implementation locally?
- [x] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

File: src/components/ValidationErrorsCard.tsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+      }
+      case 'repeatableSet': {
+        const validationData = formElementsValidation[el.name]
+        if (
+          !!validationData &&
+          typeof validationData !== 'string' &&
+          validationData.type === 'repeatableSet'
+        ) {
+          if (validationData.set) {
+            console.log(`${idPrefix}${el.name}`)
```
Comment (jdawg093) [inline src/components/ValidationErrorsCard.tsx:63]:
```suggestion
```

File: src/components/ValidationErrorsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+      }
+      case 'infoPage':
+      case 'form': {
+        const validationData = formElementsValidation[el.name]
+        if (
+          !!validationData &&
+          typeof validationData !== 'string' &&
+          validationData.type === 'formElements'
+        ) {
+          validationData.formElements
```
Comment (jdawg093) [inline src/components/ValidationErrorsCard.tsx:93]:
```suggestion
```
Assuming we want this gone?

File: src/components/ValidationErrorsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+}) => {
+  return elements.reduce<Array<ValidationErrorMetaData>>((memo, el) => {
+    switch (el.type) {
+      case 'page': {
+        memo.push(
+          ...getValidationErrors({
+            formElementsValidation,
+            elements: el.elements,
+            page: el,
+            idPrefix,
```
Comment (jdawg093) [inline src/components/ValidationErrorsCard.tsx:39]: We don't have a prefix for different pages?

Comment (jdawg093) [reply src/components/ValidationErrorsCard.tsx:39]: ah, I see now, `page` is a type here, makes sense

File: src/styles/validation-errors-card.scss [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  align-items: center;
+}
+
+.ob-validation-notification-card-header-title-text {
+  font-weight: $weight-semibold;
+}
+
+.ob-validation-notification-card-header-title-icon {
+  margin-right: 0.5rem;
+  font-size: 22px;
```
Comment (jdawg093) [inline src/styles/validation-errors-card.scss:82]: Does this (and some other hard coded values) have pre-existing variables we can use?

Comment (Zaxist) [reply src/styles/validation-errors-card.scss:82]: Mm, not really, theres existing icon size classes, but small is 18 and medium is like 30, so I couldn't think of a class name to go between the two, so I just hardcoded it.

Comment (Zaxist) [reply src/styles/validation-errors-card.scss:82]: Some of the rem stuff could probably be replaced with the `$sizeX` sass variables, though.

Comment (jdawg093) [reply src/styles/validation-errors-card.scss:82]: Happy to skip over the font-size, up to you if you want to replace any of the hardcoded rems with variables 👍

Comment (Zaxist) [reply src/styles/validation-errors-card.scss:82]: I'll add a task for it in jira.
